nucliadb_node/src/grpc/grpc_writer.rs

```diff
@@ -280,16 +280,123 @@ impl NodeWriter for NodeWriterGRPCDriver {
         Ok(tonic::Response::new(status))
     }
 
-    async fn add_vector_set(&self, _: Request<NewVectorSetRequest>) -> Result<Response<OpStatus>, Status> {
-        Err(tonic::Status::internal("Coming soon.."))
+    async fn add_vector_set(&self, request: Request<NewVectorSetRequest>) -> Result<Response<OpStatus>, Status> {
+        let span = Span::current();
+
+        let request = request.into_inner();
+        let similarity = request.similarity();
+        let normalize_vectors = request.normalize_vectors;
+        let Some(VectorSetId {
+            shard: Some(ShardId {
+                id: shard_id,
+            }),
+            vectorset,
+        }) = request.id
+        else {
+            return Ok(tonic::Response::new(OpStatus {
+                status: op_status::Status::Error.into(),
+                detail: "Vectorset ID must be provided".to_string(),
+                ..Default::default()
+            }));
+        };
+
+        let shards = Arc::clone(&self.shards);
+        let task = move || {
+            run_with_telemetry(info_span!(parent: &span, "Add a vectorset"), move || {
+                let shard = obtain_shard(shards, shard_id.clone())?;
+                shard.create_vectors_index(NewVectorsIndex {
```

Comment:
We talked about including the dimension in the create vectorset request. Are we not doing that for any particular reason?

After today's talk we would also need to eventually add more info like the datatype, etc.

Reply:
No, I probably missed the conversation. I'll add it

Reply:
No, I probably missed the conversation. Anyway, changing this would imply changing the new shard request too. Do we want to mix it in this PR?